### PR TITLE
PP forward kwargs question - old

### PR DESCRIPTION
**Pull request by nlpfollower**
Couldn't open an issue, so opening this totally random diff was the alternative.

I've been exploring your fork and saw you're adding support for the llava model -- very cool. I was also adding some flavor to this awesome repo, though only adapting it for post-training losses like dpo, and adding stuff like attention mask, sample packing, eval.

What led me to your fork was this line:
```
pp_schedule.step(target=targets, losses=losses, **{"inputs_embeds":inputs_embeds})
```

I was trying to pass an attention mask to the forward step, and using kwargs seems like the right approach. However, I found that the torch stage implementation has this tendency to flatten kwargs and add them to the inputs for gradient computation. Which seems problematic.

Maybe, do you know what's the correct way to pass no-grad forward kwargs to the torch pipeline schedule? Thanks a lot!

### CODE REVIEW

**Comment by nlpfollower** [comment]
Ignore, reopened in #3
